src/main/java/core/basesyntax/service/ReportGenerator.java

```diff
@@ -0,0 +1,11 @@
+package core.basesyntax.service;
+
+import core.basesyntax.db.FruitTransaction;
+import java.util.List;
+
+public interface ReportGenerator {
+    List<FruitTransaction> generateReport(List<FruitTransaction> listOfBalance,
```

Comment:
The method signature for `generateReport` seems too specific and might not be flexible for changes. Consider abstracting the details of the transactions and possibly use a single list of transactions or another structure that encapsulates the different transaction types.